keps/sig-network/2079-network-policy-port-range/README.md

```diff
@@ -0,0 +1,623 @@
+nitty-gritty.
+-->
+
+In NetworkPolicy specification, inside ``NetworkPolicyPort`` object struct specify a new ``PortRange`` field:
+
+```
+// NetworkPolicyPort describes a port to allow traffic on
+type NetworkPolicyPort struct {
+    PortRange *string
+}
+```
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+As the owner of a web application I want to allow the traffic to my Pods in different ports without having to create multiple network policies. As an example, my application is exposed in ports 80, 443 and 8443 and I want to group this in a single rule
+
+#### Story 2
+
+I have another application that communicates with NodePorts of a different cluster and I want to allow the egress of the traffic only the NodePort range (eg. 30000-32767) as I don't know which port is going to be allocated in the other side, but don't want to create a rule for each of them.
+
+#### Story 3
+
+As the owner of an application, I want to allow the communication with a set of ports between 6000 and 7000 except port 6667.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+CNIs will need to support the new field in their controllers. For this case we'll try to make broader communication with the main CNIs so they can be aware of the new field.
+
+The field will need some syntax validation either, to not allow the mix of named and non-named ports inside a range (like http-9000), and also to allow only numbered ports inside uint16 range.
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+API changes to NetworkPolicy:
+* Add a new field ``spec.ingress|egress.ports.PortRange
+* On field creation/change a validation should happen to verify if the format is compliant with ``nnnnn,nnnn1-nnnn5,http`` where nnnnn is a number between 0 and 65535 and named ports aren't mixed with numbered ports
```

Comment:
I don't think we've had much discussion yet on what the API for this should look like, maybe something to discuss on Monday's call?

I'm a bit opposed to putting the range semantic in the field value like `nnnn1-nnnn5`, I'm wondering if instead we might want explicit fields for `to`, `from` and `except`, something like:
```
- ingress:
  - portRange:
      from: 30000
      to: 32000
      except:
          - 31000
          - 31001
```  

Wondering how exceptions are expressed with ``nnnnn,nnnn1-nnnn5,http``, maybe I missed that.

Reply:
Yeah, I remember there was a discussion somewhere about the syntax, but because I couldn't find I've wrote whatever came to my mind expected you folks to jump in and discuss with me the best syntax :)